api/test/wfl/integration/datarepo_test.clj

```diff
@@ -77,20 +78,86 @@
             (is (= 1 row_count))
             (is (= 0 bad_row_count))))))))
 
-(def ^:private testing-dataset "3b41c460-d994-47a4-9bf7-c59861e858a6")
+(def ^:private testing-dataset "ff6e2b40-6497-4340-8947-2f52a658f561")
 
 ;; Get row-ids from BigQuery and use them to create a snapshot.
 (deftest test-create-snapshot
   (let [tdr-profile (env/getenv "WFL_TDR_DEFAULT_PROFILE")
-        {:keys [dataProject] :as dataset} (datarepo/dataset testing-dataset)
-        table     "flowcell"
-        start-datetime "2021-03-17"
-        end-datetime   "2021-03-19"
-        row-ids (->> (datarepo/make-snapshot-query dataset table start-datetime end-datetime)
-                     (bigquery/query-sync dataProject)
-                     :rows flatten)]
-    (testing "creating snapshot"
-      (fixtures/with-temporary-snapshot
-        (snapshots/unique-snapshot-request tdr-profile dataset table row-ids)
-        #(let [snapshot (datarepo/snapshot %)]
-           (is (= % (:id snapshot))))))))
+        dataset     (datarepo/dataset testing-dataset)
+        table       "flowcell"
+        from        "2021-03-30"
+        until       "2021-03-31"
+        row-ids     (-> (datarepo/query-table-between
+                         dataset
+                         table
+                         [from until]
+                         [:datarepo_row_id])
+                        :rows
+                        flatten)]
+    (fixtures/with-temporary-snapshot
+      (snapshots/unique-snapshot-request tdr-profile dataset table row-ids)
+      #(let [snapshot (datarepo/snapshot %)]
+         (is (= % (:id snapshot)))))))
+
+(deftest test-flattened-query-result
+  (let [samplesheets (-> (datarepo/dataset testing-dataset)
+                         (datarepo/query-table "flowcell" [:samplesheets])
+                         :rows
+                         (->> (mapcat first)))]
+    (is (every? string? samplesheets) "Nested arrays were not normalized.")))
+
+(defn ^:private table->map-view
+  "Create a hash-map 'view' of the BigQuery `table`. A 'view' is one"
+  [{:keys [schema rows] :as _table}]
+  (let [make-entry (fn [idx field] [(-> field :name keyword) idx])
+        key->index (into {} (map-indexed make-entry (:fields schema)))]
+    (map (util/curry #(when-let [idx (key->index %2)] (%1 idx))) rows)))
+
+(defn ^:private maps->table
+  "Transform a list of maps into a table with columns given by `attributes`."
+  [maps attributes]
+  (let [table {:schema {:fields (mapv #(-> {:name (name %)}) attributes)}}
+        f     #(reduce (fn [row attr] (conj row (% attr))) [] attributes)]
+    (assoc table :rows (map f maps))))
+
+(defn ^:private make-entity-import-request-tsv
+  [from-dataset primary-key maps]
+  ;; the columns need to be ordered such that that primary key comes first
+  (let [columns (>>> #(dissoc % primary-key) keys sort #(cons primary-key %))]
```

Comment:
we could get them from `from-dataset` or we could query the workspace for the attributes of this entity type.
In the test, we'd have to hard code them as I'm spinning up a temporary workspace.

Reply:
For `:flowcell`
```clojure 
(def columns (>>> firecloud/get-workspace
                             :flowcell
                             (juxt :idName :attributeNames)
                             #(apply cons %)))
```